tests/typings/Snapshot.test-d.ts

```diff
@@ -0,0 +1,95 @@
+import type { TypeEqual } from 'ts-expect'
+import { expectType } from 'ts-expect'
+import { INTERNAL_Snapshot as Snapshot } from '../../src/vanilla'
+
+// Converts object properties to readonly
+expectType<
+  TypeEqual<
```

Comment:
This is really neat, I hadn't seen `ts-expect` before and am going to start using it. Thanks! :-)

Reply:
There's also similar stuff in [`@type-challanges/utils`](https://www.npmjs.com/package/@type-challenges/utils) package, but I'm not sure if it intended for tests too (probably not) x)